universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
+    // if object is a ship, empire also gets knowledge of its design
+    if (vis >= VIS_PARTIAL_VISIBILITY) {
```

Comment:
could switch to `if (vis < VIS_PARTIAL_VISIBILITY) return;` and de-indent the following code